### PR TITLE
Remove privileged access from metrics-exporter

### PR DESCRIPTION
**Pull request by umangachapagain**
Fixes: [Bug 2124379](https://bugzilla.redhat.com/show_bug.cgi?id=2124379)

Depends on #1812

### CODE REVIEW

File: deploy/bundle/manifests/ceph-conf.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+data:
+  ceph.conf: |
+    [global]
+    auth_cluster_required = cephx
+    auth_service_required = cephx
+    auth_client_required = cephx
+  # keyring is a required key and its value should be empty
+  keyring: |
+metadata:
+  name: ocs-metrics-exporter-ceph-conf
```
Comment (Madhu-1) [inline deploy/bundle/manifests/ceph-conf.yaml:12]: GitHub is complaining about no new line at the end of the file.

File: deploy/bundle/manifests/ocs-operator.clusterserviceversion.yaml [hunk 2]
```diff
@@ -3042,13 +3042,30 @@ spec:
...
                 operator: Equal
                 value: "true"
+              volumes:
+              - configMap:
+                  items:
+                  - key: ceph.conf
+                    path: ceph.conf
+                  - key: keyring
+                    path: keyring
+                  name: ocs-metrics-exporter-ceph-conf
```
Comment (Madhu-1) [inline deploy/bundle/manifests/ocs-operator.clusterserviceversion.yaml:3067]: no need to mount each keys, just mount the whole configmap like this https://github.com/ceph/ceph-csi/commit/cc6c51395ebb41550a535caf952abc656caf44f6?

File: rbac/ceph-conf.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+data:
+  ceph.conf: |
+    [global]
+    auth_cluster_required = cephx
+    auth_service_required = cephx
+    auth_client_required = cephx
+  # keyring is a required key and its value should be empty
+  keyring: |
+metadata:
+  name: ocs-metrics-exporter-ceph-conf
```
Comment (Madhu-1) [inline rbac/ceph-conf.yaml:12]: Require new line at end of the file

File: controllers/ocsinitialization/sccs.go [hunk 4]
```diff
@@ -176,38 +175,3 @@ func newNooBaaEndpointSCC(namespace string) *secv1.SecurityContextConstraints {
 
 	return scc
 }
-
-func newMetricsExporterSCC(namespace string) *secv1.SecurityContextConstraints {
```
Comment (Madhu-1) [inline controllers/ocsinitialization/sccs.go:180]: @umangachapagain why SCC is removed? Dont we need SCC for metrics exporter?

File: deploy/bundle/manifests/ceph-conf.yaml (hunk 1 above)
Comment (Madhu-1) [inline deploy/bundle/manifests/ceph-conf.yaml:12]: just checking, we dont need a namespace here, right?

File: metrics/Dockerfile [hunk 5]
```diff
@@ -1,4 +1,4 @@
-FROM golang:1.17 as builder
+FROM golang:1.18 as builder
```
Comment (iamniting) [inline metrics/Dockerfile:1]: This one is already merged, We can drop this commit

**Review by iamniting — CHANGES_REQUESTED** [review]

File: controllers/ocsinitialization/sccs.go (hunk 4 above)
Comment (umangachapagain) [reply controllers/ocsinitialization/sccs.go:180]: Not anymore. Earlier we needed it for some extra permissions, but now it's working without those.

File: deploy/bundle/manifests/ceph-conf.yaml (hunk 1 above)
Comment (umangachapagain) [reply deploy/bundle/manifests/ceph-conf.yaml:12]: No, it should be created in a namespace where bundle is installed. Namespace is not allowed here.

**Review by Madhu-1 — APPROVED** [review]

**Review by iamniting — APPROVED** [review]
/lgtm

**Pull request merged**